scripts/scil_convert_json_to_xlsx.py

```diff
@@ -0,0 +1,418 @@
+
+    dataframes = []
+    df_names = []
+    index = ['Population']
+    for metric_id, metric_name in enumerate(metrics):
+        dataframes.append(pd.DataFrame(data=np.array([means[:, metric_id]]),
+                                       index=index,
+                                       columns=column_names))
+        df_names.append(metric_name + "_mean")
+
+        dataframes.append(pd.DataFrame(data=np.array([stddev[:, metric_id]]),
+                                       index=index,
+                                       columns=column_names))
+        df_names.append(metric_name + "_std")
+
+    return dataframes, df_names
+
+
+def _create_xlsx_from_json(json_path, xlsx_path,
+                           sort_subs=True, sort_bundles=True,
+                           ignored_bundles_fpath=None,
+                           stats_over_population=False):
+    with open(json_path, 'r') as json_file:
+        stats = json.load(json_file)
+
+    subs = list(stats.keys())
+    if sort_subs:
+        subs.sort()
+
+    bundle_names = _get_all_bundle_names(stats)
+    if sort_bundles:
+        bundle_names.sort()
+
+    if ignored_bundles_fpath is not None:
+        with open(ignored_bundles_fpath, 'r') as f:
+            bundles_to_ignore = [l.strip() for l in f]
+        bundle_names = filter(lambda name: name not in bundles_to_ignore,
+                              bundle_names)
+
+    cur_stats_func = _get_stats_parse_function(stats, stats_over_population)
+
+    dataframes, df_names = cur_stats_func(stats, subs, bundle_names)
+
+    if len(dataframes):
+        _write_dataframes(dataframes, df_names, xlsx_path)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_json',
+                   help='File containing the json stats')
+
+    p.add_argument('out_xlsx',
+                   help='Output Excel file for the stats.')
+
+    p.add_argument('--no_sort_subs', action='store_false',
+                   help='If set, subjects won\'t be sorted alphabetically.')
+
+    p.add_argument('--no_sort_bundles', action='store_false',
+                   help='If set, bundles won\'t be sorted alphabetically.')
+
+    p.add_argument('--ignore_bundles', metavar='FILE',
+                   help='Path to a text file containing a list of bundles '
+                        'to ignore')
+
+    p.add_argument('--stats_over_population', action='store_true',
+                   help='If set, consider the input stats to be over an '
+                        'entire population and not subject-based')
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_json)
+    assert_outputs_exist(parser, args, args.out_xlsx)
+
+    _create_xlsx_from_json(args.in_json, args.out_xlsx,
```

Comment:
maybe remove all _ before the name of functions...These are not private or related to a class.
+ docstring for everybody !

Reply:
These functions are quite cryptic, really specific, any chance the docstring could be done by the person who did them?